src/common/utils_proto.h

```diff
@@ -263,6 +266,46 @@ namespace fb_utils
 	// Frequently used actions with clumplets
 	bool isBpbSegmented(unsigned parLength, const unsigned char* par);
 
+
+	// Put integer value into info buffer
+	template<typename T>
+	inline unsigned char* putInfoItemInt(const unsigned char item, T value,
+		unsigned char* ptr, const unsigned char* end)
+	{
+		static_assert(std::is_integral_v<T>, "Integral type expected");
+
+		constexpr auto len = sizeof(T);
+		static_assert(len == 1 || len == 2 || len == 4 || len == 8, "unknown data type");
+
+		if (ptr + len + 1 + 2 > end)
+		{
+			if (ptr < end)
+			{
+				*ptr++ = isc_info_truncated;
+				if (ptr < end)
+					*ptr++ = isc_info_end;
+			}
+			return nullptr;
+		}
+
+		*ptr++ = item;
+		*ptr++ = len;
+		*ptr++ = 0;
+
+		if (len == 8)
+			put_vax_int64(ptr, value);
+		else if (len == 4)
+			put_vax_long(ptr, value);
+		else if (len == 2)
+			put_vax_short(ptr, value);
+		else if (len == 1)
+			*ptr = value;
```

Comment:
I'm a bit too late, but maybe we can use `if constexpr/else if constexpr` to evaluate this condition at compile time (as I can see `len` is constexpr variable, so this should work)?
Note: This is C++17 feature.

Reply:
Done, thanks